www/js/render-solution.js

```diff
@@ -600,12 +627,45 @@ define([
 				(!this.authorSolution || !this.authorSolution.status.error))
 				this.tabContainer.selectChild(selectedTab);
 			this.dialogWindow.show();
+			var graphHelpButton = dom.byId('graphHelpButton');
+			console.log("graph help shown",this._model.getGraphHelpShown());
+			if(!this._model.getGraphHelpShown()&&graphHelpButton ) {
+				domClass.add(graphHelpButton, "glowNode");
+				this._model.setGraphHelpShown(true);
+				this.state.put("isGraphHelpShown",true);
+			}
 		},
 
 		hide: function(){
 			//stub for logging graph closing event
 			dom.byId("graphErrorMessage").innerHTML = "";
 			dom.byId("solutionMessage").innerHTML = "";
+			var problemComplete = this._model.matchesGivenSolutionAndCorrect();
+			var problemDoneHint;
+			if(problemComplete){
+				if(!this._model._isDoneMessageShown) {
+					problemDoneHint = new tooltipDialog({
+						style: "width: 300px;",
+						content: '<p>'+this._messages['problemComplete']+'</p>' +
+							' <button type="button" data-dojo-type="dijit/form/Button" id="closeHint">Ok</button>',
+						onShow: function () {
+							on(registry.byId('closeHint'), 'click', function () {
+								console.log("clicked prob done hint closed");
+								popup.close(problemDoneHint);
+							});
+						},
+						onBlur: function(){
+							popup.close(problemDoneHint);
+						}
+					});
+					popup.open({
+						popup: problemDoneHint,
+						around: dom.byId('doneButton')
+					});
+					this._model._isDoneMessageShown = true;
+					this.state.put("isDoneMessageShown",true);
```

Comment:
please Dharani the idea is not to just write the code. But their has to be a structure to it. This just makes it hard to manage in the future. The same comments here.

Reply:
@dharanisakary please respond to this and other review comments here.